From 3124f67048c0ac7397f24ca8a82a0aa615b1d684 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Mon, 9 Jan 2023 03:36:03 +0100
Subject: [PATCH] apt: Display more useful pretty names for repositories

Previously they all had names like "Xenial" or "Stable", which is not
very useful and causes users to blindly disable repositories that they
might need.
---
 backends/apt/apt-sourceslist.cpp | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/backends/apt/apt-sourceslist.cpp b/backends/apt/apt-sourceslist.cpp
index c0d3a94c7..7c06d741d 100644
--- a/backends/apt/apt-sourceslist.cpp
+++ b/backends/apt/apt-sourceslist.cpp
@@ -472,7 +472,24 @@ string SourcesList::SourceRecord::niceName()
         ret += " Sources";
     }
 
-    return ret;
+    std::string uri_info;
+    size_t schema_pos = URI.find("://");
+    if (schema_pos == std::string::npos) {
+        uri_info = URI;
+    } else {
+        uri_info = URI.substr(schema_pos + 3);
+        if (uri_info.back() == '/')
+            uri_info.pop_back();
+    }
+
+    if (g_pattern_match_simple ("*.debian.org/*", uri_info.c_str()))
+        return "Debian " + ret;
+    if (g_pattern_match_simple ("*.ubuntu.com/*", uri_info.c_str()))
+        return "Debian " + ret;
+    if (g_pattern_match_simple ("*.pureos.net/*", uri_info.c_str()))
+        return "PureOS " + ret;
+
+    return uri_info + " - " + ret;
 }
 
 string SourcesList::SourceRecord::repoId()
